examples/03_connectivity/plot_ica_neurovault.py

```diff
@@ -0,0 +1,148 @@
+# Here by default we only download 80 images to save time,
+# but for better results I recommend using at least 200.
+print("Fetching Neurovault images; "
+      "if you haven't downloaded any Neurovault data before "
+      "this will take several minutes.")
+nv_data = fetch_neurovault(max_images=80, fetch_neurosynth_words=True)
+
+images = nv_data['images']
+term_weights = nv_data['word_frequencies']
+vocabulary = nv_data['vocabulary']
+
+# Clean and report term scores
+term_weights[term_weights < 0] = 0
+total_scores = np.mean(term_weights, axis=0)
+
+print("\nTop 10 neurosynth terms from downloaded images:\n")
+print('{2:>20}{0:>15} : {1:<15}\n{2:>20}{2:->30}'.format(
+    'term', 'total score', ''))
+for term_idx in np.argsort(total_scores)[-10:][::-1]:
+    print('{0:>35} : {1:.3f}'.format(
+        vocabulary[term_idx], total_scores[term_idx]))
+
+
+######################################################################
+# Reshape and mask images
+
+from nilearn.datasets import load_mni152_brain_mask
+from nilearn.input_data import NiftiMasker
+
+print("\nReshaping and masking images.\n")
+
+with warnings.catch_warnings():
+    warnings.simplefilter('ignore', UserWarning)
+    warnings.simplefilter('ignore', DeprecationWarning)
+
+    mask_img = load_mni152_brain_mask()
+    masker = NiftiMasker(
+        mask_img=mask_img, memory='nilearn_cache', memory_level=1)
+    masker = masker.fit()
+
+    # Images may fail to be transformed, and are of different shapes,
+    # so we need to transform one-by-one and keep track of failures.
+    X = []
+    is_usable = np.ones((len(images),), dtype=bool)
+
+    for index, image_path in enumerate(images):
+        image = clean_img(image_path)
+        try:
+            X.append(masker.transform(image))
+        except Exception as e:
+            meta = nv_data['images_meta'][index]
+            print("Failed to mask/reshape image: id: {0}; "
+                  "name: '{1}'; collection: {2}; error: {3}".format(
+                      meta.get('id'), meta.get('name'),
+                      meta.get('collection_id'), e))
+            is_usable[index] = False
+
+# Now reshape list into 2D matrix, and remove failed images from terms
+X = np.vstack(X)
+term_weights = term_weights[is_usable, :]
+
+
+######################################################################
+# Run ICA and map components to terms
+
+print("Running ICA; may take time...")
+n_components = 40
+fast_ica = FastICA(n_components=n_components, random_state=0)
+ica_maps = fast_ica.fit_transform(X.T).T
+
+term_weights_for_components = np.dot(fast_ica.components_, term_weights)
+print('Done, plotting results.')
+
+
+######################################################################
+# Generate figures
+
+from nilearn import plotting
+
+old_max_open = plotting.img_plotting.matplotlib.pyplot.rcParams[
+    'figure.max_open_warning']
+plotting.img_plotting.matplotlib.pyplot.rcParams[
+    'figure.max_open_warning'] = n_components + 2
```

Comment:
I would remove this. This is far too advanced. I would prefer a more elaborated figure with subplots rather than 40 plots.